src/main/java/com/mraof/minestuck/blockentity/IntellibeamLaserstationBlockEntity.java

```diff
@@ -0,0 +1,237 @@
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class IntellibeamLaserstationBlockEntity extends BlockEntity
+{
+	public static final String DECODING_PROGRESS = "block.minestuck.intellibeam_laserstation.decoding_progress";
+	public static final String CAPTCHA_DECODED = "block.minestuck.intellibeam_laserstation.captcha_decoded";
+	
+	private static final int EXP_LEVEL_CAPACITY = 10;
+	
+	private final Map<String, Integer> storedDecodings = new HashMap<>();
+	
+	private ItemStack analyzedCard = ItemStack.EMPTY;
+	private int waitTimer = 0;
+	
+	public IntellibeamLaserstationBlockEntity(BlockPos pos, BlockState state)
+	{
+		super(MSBlockEntityTypes.INTELLIBEAM_LASERSTATION.get(), pos, state);
+	}
+	
+	public static void serverTick(Level level, BlockPos pos, BlockState state, IntellibeamLaserstationBlockEntity intellibeam)
+	{
+		if(intellibeam.waitTimer > 0)
+		{
+			intellibeam.waitTimer--;
+		}
+	}
+	
+	public void onRightClick(Player player)
+	{
+		if(level != null && !level.isClientSide)
+		{
+			ItemStack heldItem = player.getMainHandItem();
+			ItemStack cardWithItem = AlchemyHelper.getDecodedItem(heldItem);
+			
+			if(waitTimer > 0)
+			{
+				return;
+			}
+			
+			if(!analyzedCard.isEmpty() && player.isShiftKeyDown())
+			{
+				takeCard(player);
+				waitTimer = 10;
+			} else if(AlchemyHelper.isReadableCard(analyzedCard))
+			{
+				this.level.playSound(null, this.worldPosition, MSSoundEvents.INTELLIBEAM_LASERSTATION_REMOVE_CARD.get(), SoundSource.BLOCKS, 0.5F, 0.1F);
+				player.displayClientMessage(Component.translatable(CAPTCHA_DECODED), true);
+				waitTimer = 10;
+			} else if(analyzedCard.isEmpty() && !AlchemyHelper.isReadableCard(analyzedCard) && cardWithItem.is(MSTags.Items.UNREADABLE))
+			{
+				setCard(heldItem.split(1));
+				waitTimer = 10;
+			} else if(getCardItemExperience() >= EXP_LEVEL_CAPACITY)
+			{
+				decodeAndEjectCard(player);
+			} else
+			{
+				addExperience(player);
+				waitTimer = 10;
+			}
+		}
+	}
+	
+	private Integer getCardItemExperience()
+	{
+		return storedDecodings.getOrDefault(getCardItemName(), 0);
+	}
+	
+	private String getCardItemName()
+	{
+		return CardCaptchas.getRegistryNameFromItem(AlchemyHelper.getDecodedItem(analyzedCard).getItem());
+	}
+	
+	public void decodeAndEjectCard(Player player)
+	{
+		applyDecodedTag(analyzedCard);
+		takeCard(player);
+		
+		waitTimer = 10;
+	}
+	
+	public String processExperienceGuage()
+	{
+		int numberOfBars = "▯▯▯▯▯▯▯▯▯▯".length(); //used to visualize how many experience bars there are
+		
+		char[] stringChars = new char[numberOfBars];
+		
+		for(int i = 0; i < numberOfBars; i++)
+		{
+			if(i < getCardItemExperience())
+				stringChars[i] = '▮';
+			else
+				stringChars[i] = '▯';
+		}
+		
+		return '<' + new String(stringChars) + '>'; //if two levels of experience have been invested, it will look like this: "<▮▮▯▯▯▯▯▯▯▯>"
```

Comment:
In game, the bar characters are smaller than the text. Not sure if there is a simple solution to make them similarly sized like how they appear in Github and IntelliJ